src/main/java/com/ibm/watson/modelmesh/ModelMesh.java

```diff
@@ -244,29 +225,26 @@ public abstract class ModelMesh extends ThriftService
     public final long LOCAL_JANITOR_FREQ_SECS = Long.getLong("tas.janitor_freq_secs", 6 * 60); // 6mins
 
     // interval for rate checking task which tracks rate of reqs to each model
-    protected final long RATE_CHECK_INTERVAL_MS = Long.getLong("tas.ratecheck_freq_ms", 11_000L); // 11sec
+    protected final long RATE_CHECK_INTERVAL_MS = Long.getLong("tas.ratecheck_freq_ms", 10_000L); // 10sec
 
     protected static final int DEFAULT_SCALEUP_RPM = 2000;
 
-    /* If single-copy models are seen by the janitor to have been used
-     * within these times, it will ensure two copies are loaded (in
-     * different instances). A different "age" relative to the model's
-     * lastUsed cache timestamp is used depending on whether any "real"
-     * usage has been recorded yet for the model in this instance.
-     * The smaller time for the "unused" case is to avoid scaling up
-     * newly-added models which have a recent cache timestamp but have
-     * never actually been used.
-     * The times should be greater than the janitor frequency
+    /* These are the "ages" in terms of iterations of the rateTrackingTask
+     * between which a prior use of a single-copy model will trigger a second
+     * copy to be loaded. With the default rate checking task frequency of 10 sec
+     * they correspond to 40 and 7 minutes respectively.
+     * This is a heuristic for identifying "regular" usage as apposed to one-off
+     * or short burst.
      */
-    public static final long SECOND_COPY_ADD_AGE_UNUSED_MS = 10 * 60_000L; // 10mins
-    public static final long SECOND_COPY_ADD_AGE_USED_MS = 12 * 3600_000L; // 12hours
+    final int SECOND_COPY_MAX_AGE_ITERS = Integer.getInteger("mm.max_second_copy_age_itrs", 240);
+    final int SECOND_COPY_MIN_AGE_ITERS = Integer.getInteger("mm.min_second_copy_age_itrs", 42);
 
     /* Last-used age after which second model copies will be removed
      * (2->1 scaledown). Note this is a maximum and a smaller value might
-     * be used if the global age of the cache is small (less than 5x this value).
+     * be used if the global age of the cache is small (less than 10x this value).
```

Comment:
Should this heuristic also be configurable? For the large model use case, users expect the global cache age to be in days, and it would be a really bad thing if it ever went down to 1 hour.

I guess the rest of this fix aims to avoid the 1->2 scaleup to help the global cache age stay long in the first place, but in the case where enough of these larger models are scaled up to 2 copies anyway, maybe it would be useful to configure this value of `small` so that it could be set to like... 12 hours? 24 hours?

Or, maybe it would be enough just to make this value itself configurable, so users could more aggressively clean out second copies where they know models are large and cache space is at a premium?

Reply:
I'm hoping this is a reasonable balance for now, we can always revisit exposing additional levers if it turns out to be needed.